### PR TITLE
Add sections for allegation details

### PR DESCRIPTION
**Pull request by vassyz**
https://trello.com/c/vOVS2gZF/1345-theallegationsection

### CODE REVIEW

File: app/forms/referrals/sections/allegation_section.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+module Referrals
+  module Sections
+    class AllegationSection < Section
+      def items
+        [
+          Referrals::Allegation::DetailsForm.new(referral:),
+          Referrals::Allegation::ConsiderationsForm.new(referral:),
```
Comment (paulodeon) [inline app/forms/referrals/sections/allegation_section.rb:7]: I think this should be `!referral.from_member_of_public? && Referrals::Allegation::ConsiderationsForm.new(referral:)`

File: app/forms/referrals/sections/evidence_section.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+module Referrals
+  module Sections
+    class EvidenceSection < Section
+      def items
+        [
+          referral.evidences.none? && Referrals::Evidence::StartForm.new(referral:),
+          referral.evidences.none? && Referrals::Evidence::UploadForm.new(referral:),
```
Comment (paulodeon) [inline app/forms/referrals/sections/evidence_section.rb:7]: I think this should be something like this `referral.evidences.none? && referral.has_evidence == 'yes' && Referrals::Evidence::UploadForm.new(referral:),` although it wont matter until later

**Comment by vassyz** [comment]
@paulodeon can you please check if these changes look alright: https://github.com/DFE-Digital/refer-serious-misconduct/pull/688/commits/3cba11247a04b6e6e43e18f0235b54d83a65834f

There was also a duplicate `DbsForm` row I had to remove. 

I'm not sure what the best approach in terms of readability is. I could switch to using `if/append`, similar to @tavy88 's https://github.com/DFE-Digital/refer-serious-misconduct/pull/687/files#diff-8e89a62742cba6190320db2a0e48414c8bff84882ccc2998fda473cefe9f0ad5

**Comment by paulodeon** [comment]
Looks good - just a missing `compact_blank`.  

Personally I prefer your approach in this case for readability - especially if there are multiple conditionals

**Review by tavy88 — APPROVED** [review]
LGTM - I guess a round of manual tests would be good.

File: app/forms/referrals/sections/allegation_section.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  module Sections
+    class AllegationSection < Section
+      def items
+        [
+          Referrals::Allegation::DetailsForm.new(referral:),
+          referral.from_member_of_public? &&
+            Referrals::Allegation::ConsiderationsForm.new(referral:),
+          referral.from_employer? && Referrals::Allegation::DbsForm.new(referral:),
+          Referrals::Allegation::CheckAnswersForm.new(referral:)
+        ]
```
Comment (paulodeon) [inline app/forms/referrals/sections/allegation_section.rb:11]: Just need a `.compact_blank` here

**Review by paulodeon — APPROVED** [review]

**Comment by vassyz** [comment]
[TheAllegationSection](https://trello.com/c/vOVS2gZF/1345-theallegationsection)

**Pull request merged**